src/openloco/industry.cpp

```diff
@@ -158,20 +161,24 @@ namespace openloco
 
     void industry::sub_453354()
     {
-        auto obj = object();
+        // 0x00453366
         int16_t tmp_a = var_DB / 16;
         int16_t tmp_b = std::max(0, var_DD - tmp_a);
         int16_t tmp_c = var_DB - tmp_b;
         int16_t tmp_d = std::min(tmp_c / 25, 255);
+
+        auto obj = object();
         if (tmp_d < obj->var_EB)
         {
-            var_DF = tmp_d / obj->var_EB;
+            var_DF = ((tmp_d << 8) / obj->var_EB) & 0xFF;
```

Comment:
id prefer this to be `*256`

Reply:
So the following?
```suggestion
            var_DF = ((tmp_d * 256) / obj->var_EB) & 0xFF;
```

We shift around a bunch in other places, though. Any particular reason? (Endianness?)

Reply:
Endianness doesn't affect operations like this. I think until we actually know what any of these variables and numbers represents it would be better to keep.

Reply:
The original doesn't actually shift; it replaces `ah` with `al` and then `xor`s `al`. We could go either way.

Reply:
Okay, I think multiply is fine then as it looks like it is scaling it up to increase precision before the divide.